src/components/filter/Filter.jsx

```diff
@@ -0,0 +1,168 @@
+          <div className="slider-container">
+            <div className="flex-row justify-spacebw text-sm">
+              <span></span>
+              <span>0</span>
+              <span>20k</span>
+              <span>40k</span>
+              <span>60k</span>
+              <span>80k</span>
+              <span>100k</span>
+            </div>
+            <input
+              type="range"
+              min="0"
+              max="100000"
+              step="20000"
+              value={byPrice}
+              onChange={(event) =>
+                dispatch({ type: "PRICE", payload: event.target.value })
+              }
+            />
+          </div>
+        </div>
+        <div className="filter-list">
+          <p className="fw-700">Category</p>
+          <div>
+            <input
+              type="checkbox"
+              id="smartphone"
+              value="smartphone"
+              checked={byCategories.smartphone}
+              onChange={() => dispatch({ type: "SMARTPHONE" })}
+            />
+            <label htmlFor="smartphone">Smartphone</label>
+          </div>
+          <div>
+            <input
+              type="checkbox"
+              id="laptop"
+              value="laptop"
+              checked={byCategories.laptop}
+              onChange={() => dispatch({ type: "LAPTOP" })}
+            />
+            <label htmlFor="laptop">Laptop</label>
+          </div>
+          <div>
+            <input
+              type="checkbox"
+              id="camera"
+              value="camera"
+              checked={byCategories.camera}
+              onChange={() => dispatch({ type: "CAMERA" })}
+            />
+            <label htmlFor="camera">Camera</label>
+          </div>
+          <div>
+            <input
+              type="checkbox"
+              id="headphones"
+              value="headphones"
+              checked={byCategories.headphones}
+              onChange={() => dispatch({ type: "HEADPHONES" })}
+            />
+            <label htmlFor="headphones">Headphones</label>
+          </div>
+        </div>
+        <div className="filter-list">
+          <p className="fw-700">Rating</p>
+          <div>
+            <input
+              type="radio"
+              id="four"
+              name="Rating"
+              value="4"
+              checked={byRating === "4"}
+              onChange={(event) =>
+                dispatch({ type: "RATING", payload: event.target.value })
+              }
+            />
+            <label htmlFor="four">4 stars & above</label>
+          </div>
+          <div>
+            <input
+              type="radio"
+              id="three"
+              name="Rating"
+              value="3"
+              checked={byRating === "3"}
+              onChange={(event) =>
+                dispatch({ type: "RATING", payload: event.target.value })
+              }
+            />
+            <label htmlFor="three">3 Stars & above</label>
+          </div>
```

Comment:
This seem like a repetitive code.  Why not map this through an array of objects

Reply:
Will try to do that. Thanks for the review